v3/setup/etcdctl-auth.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+source /etc/environment
+
+#Disable authentication
+
+#curl  -L http://127.0.0.1:2379/v2/auth/enable -XDELETE
+
+etcdctl set /etcdctl/config/root-user root
+etcdctl set /etcdctl/config/read-user read-user
+etcdctl set /etcdctl/config/write-user write-user
+
+ROOT_USERNAME=$(etcdctl get /etcdctl/config/root-user)
+ROOT_PASSWORD=$(etcdctl get /etcdctl/config/root-password)
+READ_USERNAME=$(etcdctl get /etcdctl/config/read-user)
+READ_PASSWORD=$(etcdctl get /etcdctl/config/read-password)
+WRITE_USERNAME=$(etcdctl get /etcdctl/config/write-user)
+WRITE_PASSWORD=$(etcdctl get /etcdctl/config/write-password)
+
+CRED_DIR="/opt/etcdctl"
+if [[ ! -d $CRED_DIR ]]; then
+    sudo mkdir $CRED_DIR -p
+fi
+
+#add a root user
+
+sudo echo '{"user":"'$(eval echo $ROOT_USERNAME)'", "password":"'$(eval echo $ROOT_PASSWORD)'"}' > $CRED_DIR/root-user.json
+curl  -L http://127.0.0.1:2379/v2/auth/users/$(eval echo $ROOT_USERNAME) -XPUT -d "@$CRED_DIR/root-user.json"
+
+
+#add a non root read only user
+sudo echo '{"user":"'$(eval echo $READ_USERNAME)'", "password":"'$(eval echo $READ_PASSWORD)'"}' > $CRED_DIR/read-user.json
+curl   -L http://127.0.0.1:2379/v2/auth/users/$(eval echo $READ_USERNAME) -XPUT -d "@$CRED_DIR/read-user.json"
+
+#add a non root readwrite user
+sudo echo '{"user":"'$(eval echo $WRITE_USERNAME)'", "password":"'$(eval echo $WRITE_PASSWORD)'"}' > $CRED_DIR/write-user.json
+curl  -L http://127.0.0.1:2379/v2/auth/users/$(eval echo $WRITE_USERNAME) -XPUT -d "@$CRED_DIR/write-user.json"
+
+#Create a read only role:
+
+etcdctl role add readonlyrole
```

Comment:
are there any restricted characters on role names? That is, can we just do something like

```
etcdctl role add read-only
```

2 issues with the current state:

1. mashing the words together isn't really readable 
2. `read-only-role` has the word `role` in it. It's already evident, on setting & ls'ing roles that this "thing" is already a role, so the word `role` itself does not have to be in the name